feat : 카테고리 수정 API 추가

beetrbgus opened the pull request:
## 📃 관련 이슈
- #35 
- #41 의 Merge가 선행 되어야 함.
## ✨ 변경 내용
- 변경 사항 간단 요약

## 🔎 참고 사항 (선택)
- 추가 설명, 참고 링크 등

beetrbgus committed:
feat : 수정 API 추가

beetrbgus committed:
feat : 수정 API 추가

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/application/command/usecase/category/impl/UpdateCategoryUseCaseImpl.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Override
+    public void handle(Long id, UpdateCategoryDto commandDto) {
+        Category category = queryRepository.findById(id).orElseThrow(() -> new CategoryException(CategoryErrorCode.CATEGORY_NOT_FOUND));
```
R: .orElseThrow 부분 개행 부탁드립니닷

beetrbgus committed:
Merge remote-tracking branch 'origin/feat/35-update' into feat/35-update

beetrbgus committed:
feat : 문서 추가

beetrbgus committed:
feat : 문서 추가

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/application/command/usecase/category/impl/UpdateCategoryUseCaseImpl.java:
```diff
@@ -0,0 +1,52 @@
+                    commandDto.description(), commandDto.level(), commandDto.sortOrder(), commandDto.isActive());
+        }
+
+        commandRepository.createCategory(category);
```
C: jpa 더티체킹 되니까 save 쿼리 안 날려도 되지 않나요??

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/dto/UpdateCategoryRequest.java:
```diff
@@ -0,0 +1,31 @@
+import jakarta.validation.constraints.NotBlank;
+
+public record UpdateCategoryRequest(
+        @Schema(name = "그룹 코드", example = "MENU", requiredMode = Schema.RequiredMode.REQUIRED)
```
R: 여기도 `@NotBlank` 만 사용하시면 될 것 같아오~

beetrbgus committed:
Merge remote-tracking branch 'origin/feat/35-update' into feat/35-update

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryResource.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Override
+    @PutMapping("/{id}")
```
R: `categoryId`!!
그리고 `@PathVarialbe` 빠졌슴다

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryResource.java:
```diff
@@ -0,0 +1,46 @@
+    @Override
+    @PutMapping("/{id}")
+    public void update(Long id, UpdateCategoryRequest updateRequest) {
+        UpdateCategoryDto updateCategoryDto = new UpdateCategoryDto(updateRequest.groupCode(),
```
R: 컨벤션-UpdateCategoryDto.of(request)!

beetrbgus committed:
feat : feed back 반영

Jiihyun commented on showhive-domain/domain/src/main/java/com/showhive/category/domain/Category.java:
```diff
@@ -0,0 +1,101 @@
+        this.level = level;
+        this.sortOrder = sortOrder;
+        this.isActive = isActive;
+    }
```
R: 메서드명 조금 더 구체적으로 부탁드립니다!

beetrbgus replied:
변경 완룟

Jiihyun reviewed (commented):
리뷰 반영 부탁드림니다🙇🏻‍♀️

whitejh commented on showhive-boot/admin/src/main/java/com/showhive/admin/application/command/dto/category/UpdateCategoryDto.java:
r : static 메서드에서 sortOrder를 request.level()로 넣으셨는데 오타같아요!
request.sortOrder()로 수정해야될 것 같아요

whitejh replied:
<img width="694" height="979" alt="image" src="https://github.com/user-attachments/assets/f7bbba99-0616-4553-a85b-cd3a828c5449" />

c : 쿠폰에서 commandDto를 CouponRequest로 이름 변경 필요하다고 리뷰 달아주셨는데, 카테고리쪽도 CreateCategoryRequest 이런식으로 변경해야될까요? 서로 통일이 필요할 것 같습니다.

Jiihyun replied:
준현님 코드에 리뷰 남겨주신 건 레이어 착각하신 것 같아요.
이름 변경 안 해주셔도 되고, 
컨트롤러 레이어 쪽은 request, 서비스 레이어 쪽은 dto로 하시면 될 것 같습니다!

whitejh replied:
네 확인했습니다!

whitejh commented on showhive-boot/admin/src/main/java/com/showhive/admin/application/command/dto/venue/SeatDto.java:
r : static 메서드에서 sortOrder를 request.level()로 넣으셨는데 오타같아요!
request.sortOrder()로 수정해야될 것 같아요

beetrbgus committed:
Merge branch 'refs/heads/develop' into feat/35-update

# Conflicts:
#	showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryFacade.java
#	showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryResource.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/category/query/CategoryQueryJpaRepository.java

beetrbgus committed:
refactor : 카테고리 메서드 명 변경

##############################
## 타입 설정                  ##
##############################

## feat: 기능 구현
## refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
## fix: 버그 또는 오류 해결
## style: CSS 등 스타일 관련 변경
## test: 테스트 코드 추가 또는 수정
## docs: README.md 등 문서 관련 변경
## chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
## build: CI/CD 관련 변경

beetrbgus committed:
Merge branch 'develop' of https://github.com/show-hive/showhive-backend into feat/35-update

# Conflicts:
#	showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryFacade.java
#	showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryResource.java
#	showhive-domain/domain/src/main/java/com/showhive/category/domain/Category.java

beetrbgus committed:
refactor : 메서드 명 변경

##############################
## 타입 설정                  ##
##############################

## feat: 기능 구현
## refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
## fix: 버그 또는 오류 해결
## style: CSS 등 스타일 관련 변경
## test: 테스트 코드 추가 또는 수정
## docs: README.md 등 문서 관련 변경
## chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
## build: CI/CD 관련 변경

beetrbgus committed:
refactor : createNode 개선
- 객체 생성 시 구조 수정

##############################
## 타입 설정                  ##
##############################

## feat: 기능 구현
## refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
## fix: 버그 또는 오류 해결
## style: CSS 등 스타일 관련 변경
## test: 테스트 코드 추가 또는 수정
## docs: README.md 등 문서 관련 변경
## chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
## build: CI/CD 관련 변경

beetrbgus committed:
feat : PathVariable 추가

# #############################
# # 타입 설정                  ##
# #############################

# # feat: 기능 구현
# # refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# # fix: 버그 또는 오류 해결
# # style: CSS 등 스타일 관련 변경
# # test: 테스트 코드 추가 또는 수정
# # docs: README.md 등 문서 관련 변경
# # chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# # build: CI/CD 관련 변경
